src/pytesseract.py

```diff
@@ -301,6 +301,34 @@ def image_to_string(image,
     return run_and_get_output(*args)
 
 
+def image_to_pdf_or_hocr(image,
+                    lang=None,
+                    config='',
+                    nice=0,
+                    boxes=False,
+                    output_type=Output.BYTES,
+                    extension='pdf'):
+    '''
+    Returns the result of a Tesseract OCR run on the provided image to string
+    '''
+    if boxes:
+        # Added for backwards compatibility
+        print('\nWarning: Argument \'boxes\' is deprecated and will be removed'
+              ' in future versions. Use function image_to_boxes instead.\n')
+        return image_to_boxes(image, lang, config, nice, output_type)
+
+    if extension not in ['pdf', 'hocr']:
+        extension = 'txt'
+    args = [image, extension, lang, config, nice]
+
+    if output_type == Output.DICT:
```

Comment:
Do we ever need the DICT output for the pdf extention?
I mean - the user will get the string representation of the pdf bytes, and I don't think that the user will be expect such output. Right?

I guess that we can leave it as is, but it will need refactoring in the future (for sure).